libs/community/langchain_community/embeddings/self_hosted.py

```diff
@@ -29,47 +43,51 @@ class SelfHostedEmbeddings(SelfHostedPipeline, Embeddings):
         .. code-block:: python
 
             from langchain_community.embeddings import SelfHostedEmbeddings
-            from transformers import AutoModelForCausalLM, AutoTokenizer, pipeline
             import runhouse as rh
 
-            gpu = rh.cluster(name="rh-a10x", instance_type="A100:1")
-            def get_pipeline():
-                model_id = "facebook/bart-large"
-                tokenizer = AutoTokenizer.from_pretrained(model_id)
-                model = AutoModelForCausalLM.from_pretrained(model_id)
-                return pipeline("feature-extraction", model=model, tokenizer=tokenizer)
-            embeddings = SelfHostedEmbeddings(
-                model_load_fn=get_pipeline,
-                hardware=gpu
-                model_reqs=["./", "torch", "transformers"],
-            )
-    Example passing in a pipeline path:
-        .. code-block:: python
+            class MyEmbeddingsClass:
+                def __init__(self, model_id: str, instruct: bool = False, device: int = 0):
+                    ... construction implementation here ...
 
-            from langchain_community.embeddings import SelfHostedHFEmbeddings
-            import runhouse as rh
-            from transformers import pipeline
-
-            gpu = rh.cluster(name="rh-a10x", instance_type="A100:1")
-            pipeline = pipeline(model="bert-base-uncased", task="feature-extraction")
-            rh.blob(pickle.dumps(pipeline),
-                path="models/pipeline.pkl").save().to(gpu, path="models")
-            embeddings = SelfHostedHFEmbeddings.from_pipeline(
-                pipeline="models/pipeline.pkl",
-                hardware=gpu,
-                model_reqs=["./", "torch", "transformers"],
-            )
+                def load_embedding_model(self):
+                    ... load_embedding_model implementation here ...
+
+                def embed_documents(self, *args: Any, **kwargs: Any) -> List[List[float]]:
+                    .. embed_documents implementation here ...
+
+            gpu = rh.cluster(name='rh-a10x', instance_type='g5.4xlarge', provider='aws')
+            gpu.run(commands=["pip install langchain"])
```

Comment:
assume is this bc we need langchain installed on the cluster before we install the env?